pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,6 +1037,86 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
+func (ctrl *Controller) createDefaultContainerRuntimeMC() error {
+	// Check if the crio-default-container-runtime config map exists in the openshift-machine-config-operator namespace
+	defaultContainerRuntimeCM, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error checking for %s config map: %w", configMapName, err)
+	}
+	// If the crio-default-container-runtime config map exists, the MC was already created, so skip creating it again.
+	if defaultContainerRuntimeCM != nil && !errors.IsNotFound(err) {
+		return nil
+	}
+
+	sel, err := metav1.LabelSelectorAsSelector(metav1.AddLabelToSelector(&metav1.LabelSelector{}, builtInLabelKey, ""))
+	if err != nil {
+		return err
+	}
+	// Find all the MachineConfigPools
+	mcpPoolsAll, err := ctrl.mcpLister.List(sel)
+	if err != nil {
+		return err
+	}
+
+	// Create the crio-default-container-runtime MC for all the available pools
+	for _, pool := range mcpPoolsAll {
```

Comment:
I forget if we had this conversation or not, one consideration here is that instead of doing it for all pools, we just do it for master and worker. Any custom pools the user may have will inherit from the worker configuration, so it's just a duplicated entry at the end of the day, and if the user removes the custom pool MC here, the custom pool will undergo no actual change (since the worker config is still taking effect) so unless there's a specific case we're targetting, doing it for just master/worker might be better?